waspc/waspls/src/Wasp/LSP/Commands/ScaffoldTsSymbol.hs

```diff
@@ -0,0 +1,258 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Wasp.LSP.Commands.ScaffoldTsSymbol
+  ( -- * Scaffold TS Symbol Command
+
+    -- Command @wasp.scaffold.ts-symbol@ appends a new function with the given
+    -- name to end of a particular file.
+    --
+    -- Scaffolding is based on the location of the external import. It looks for
+    -- the templates for each kind of JS/TS code in @data/lsp/templates/ts@.
+    -- For example, it differentiates between
+    --
+    -- @
+    -- query getAll { fn: import { getAll } from "@server/queries.js" }
+    -- @
+    --
+    -- and
+    --
+    -- @
+    -- action deleteAll { fn: import { deleteAll } from "@server/queries.ts" }
+    -- @
+    --
+    -- The former looks for a template named @query.fn.js@, the latter for
+    -- @action.fn.ts@. To add support for a new kind of JS/TS code to scaffold,
+    -- create a new file @data/lsp/templates/ts/<path>.<ext>@, where @<ext>@ is
+    -- the extension of the file that the code will be added to and @<path>@ is
+    -- the dot-separated path to the external import in the wasp code. Each
+    -- piece of the path is:
+    --
+    -- - The declaration type, as a string.
+    -- - @list@, for a value inside a list.
+    -- - @[n]@, for the @n@-th value in a tuple.
+    -- - The key of a dictionary field, as a string.
```

Comment:
I think I was imagining the situation where you modify how AppSpec looks like, and you forget to update some of the names of the files here, template files.

But I guess the approach I described also wouldn't help, since we are again describing structure of the AST, just in different manner (in the code), and change in AppSpec will still not affect that.

We don't have a place in our code where we check if files are on disk -> that was just a wild idea I wrote, let's not do that.

I do wonder, is there a way we can make this a bit more robust to possible changes in AppSpec? In theory, we could calculate these "expression paths" directly from AppSpec? Not sure how feasible is that hm.

One possible benefit of what you prototyped above is that we don't need as much comments though, right? Because it is kind of obvious from the code what is happening? So we don't have to explain the small DSL that you came up with for writing the names of the template files? Maybe that is a worthy improvement?

Reply:
> I think I was imagining the situation where you modify how AppSpec looks like.

Ah, that's a good point! I'm not sure how we would accomplish that though. (Well, I have some ideas but they seem so complex to implement to not be worth it)

> One possible benefit...is that we don't need as much comments...Maybe that is a worthy improvement?

I think I agree with that. I've changed it to be like this suggestion.